.github/workflows/bazel-lockfile.yml

```diff
@@ -0,0 +1,222 @@
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  bazel-bzlmod-lockfile-apple-silicon:
+    name: Apple Silicon
+    runs-on: namespace-profile-darwin # profile created in namespace console
+    steps:
+      - uses: actions/checkout@v4
+      - name: Bazel dry run
+        run: |
+          bazel \
+            --noworkspace_rc \
+            --bazelrc=./bazel/conf/.bazelrc.build \
+            build \
+            --config=ci --config=macos_ci \
+            --test_tag_filters="test_macos,test_macos_slow,-upload" \
+            //rs/... //publish/binaries/... \
+            --lockfile_mode=update \
+            --nobuild
+      - uses: actions/upload-artifact@v4
+        with:
+          name: bazel-module-lock-apple-silicon
+          path: ./MODULE.bazel.lock
+
+  bazel-bzlmod-lockfile-linux:
+    name: Linux
+    container:
+      image: ghcr.io/dfinity/ic-build@sha256:115daa5ad5149182bb0416cbe5730f305be3bb2f48df576bc2c23067eefce84b
+      options: >-
+        -e NODE_NAME --privileged --cgroupns host -v /cache:/cache -v /var/sysimage:/var/sysimage -v /var/tmp:/var/tmp -v /ceph-s3-info:/ceph-s3-info
+    runs-on:
+      group: zh1
+      labels: dind-large
+    needs: bazel-bzlmod-lockfile-apple-silicon
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Before script
+        id: before-script
+        shell: bash
+        run: |
+          [ -n "${NODE_NAME:-}" ] && echo "Node: $NODE_NAME"
+      - name: Login to Dockerhub
+        shell: bash
+        run: ./ci/scripts/docker-login.sh
+        env:
+          DOCKER_HUB_USER: ${{ vars.DOCKER_HUB_USER }}
+          DOCKER_HUB_PASSWORD_RO: ${{ secrets.DOCKER_HUB_PASSWORD_RO }}
+      - uses: actions/download-artifact@v4
+        with:
+          name: bazel-module-lock-apple-silicon
+
+      # Run a "build" with --nobuild so that Bazel updates the lockfile (if
+      # necessary). No targets will be built.
+      # The rest is copied from the main workflow.
+      - name: Bazel dry run
+        id: bazel-test-all
+        uses: ./.github/actions/bazel-test-all/
+        env:
+          CI_COMMIT_SHA: ${{ github.sha }}
+          CI_JOB_NAME: ${{ github.job }}
+          CI_JOB_URL: "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+          CI_PIPELINE_SOURCE: ${{ github.event_name }}
+          CI_PROJECT_DIR: ${{ github.workspace }}
+          BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+          CI_RUN_ID: ${{ github.run_id }}
+          RUSTFLAGS: "--remap-path-prefix=${CI_PROJECT_DIR}=/ic"
+          BUILDEVENT_DATASET: "github-ci-dfinity"
+        with:
+          BAZEL_COMMAND: "build"
+          BAZEL_TARGETS: "//..."
+          BAZEL_EXTRA_ARGS: '--lockfile_mode=update --nobuild'
+          BAZEL_CI_CONFIG: "--config=ci --repository_cache=/cache/bazel"
+          # check if PR title contains release and set timeout filters accordingly
```

Comment:
Was this comment accidentally copied from somewhere else since it seems out of place here?

Reply:
Like most things in this file I tried to (as much as possible) keep it consistent with ci-main.yml (basically copy pasted). Happy to remove though!